frontend/src/metabase/lib/dom.js

```diff
@@ -231,6 +231,17 @@ function isMetabaseUrl(url) {
     return true;
   }
 
+  const pathNameWithoutSubPath = getPathnameWithoutSubPath(urlPath);
+  const isPublicLink = pathNameWithoutSubPath.startsWith("/public/");
+  const isEmbedding = pathNameWithoutSubPath.startsWith("/embed/");
+  /**
+   * (metabase#38640) We don't want to use client-side navigation for public links or embedding
+   * because public app, or embed app are built using separate routes.
+   **/
+  if (isPublicLink || isEmbedding) {
```

Comment:
Please see the [PR description](https://github.com/metabase/metabase/pull/38647#issue-2129748766) for an explanation.